packages/opentelemetry-core/src/platform/node/ShutdownNotifier.ts

```diff
@@ -14,13 +14,22 @@
  * limitations under the License.
  */
 
+let shutDownListeners: Array<() => void> = [];
+process.once('SIGTERM', function () {
+  shutDownListeners.forEach(listener => listener());
+});
+
 /**
  * Adds an event listener to trigger a callback when a SIGTERM is detected in the process
  */
 export function notifyOnGlobalShutdown(cb: () => void): () => void {
-  process.once('SIGTERM', cb);
+  shutDownListeners.push(cb);
+
   return function removeCallbackFromGlobalShutdown() {
-    process.removeListener('SIGTERM', cb);
+    const i = shutDownListeners.findIndex((v) => v === cb);
```

Comment:
you can call the same function couple times which means you have to check first for previous assignment. I think it is easier to use here `indexOf` and `splice`, WDYT ?
```
export function notifyOnGlobalShutdown(cb: () => void): () => void {
  const index = shutDownListeners.indexOf(cb);
  if (index >= 0) {
    shutDownListeners.splice(index, 1);
  }
  return function removeCallbackFromGlobalShutdown() {
    shutDownListeners.splice(shutDownListeners.indexOf(cb), 1);
  });
}
```

Reply:
Then if I remove one callback the indexes will be different for all the other callbacks.

```typescript
function f1() {}
const rem1 = notifyOnGlobalShutdown(f1); // index 0

function f2() {}
const rem2 = notifyOnGlobalShutdown(f2); // index 1

rem1(); // index of f1 is now 0
rem2(); // try to remove non-existing index 1
```

Reply:
We could instead wrap in an object which has a unique reference and use a filter

Reply:
If someone calls `notifyOnGlobalShutdown` twice with the same function, should we expect it to be called twice or only once?

```typescript
const onShutdown = () => console.log("shutdown");
notifyOnGlobalShutdown(onShutdown);
notifyOnGlobalShutdown(onShutdown);

// How many times do we want "shutdown" to be logged?
```

Reply:
you should mimic the behaviour of "on" "once", which in this  case would log shutdown once only

Reply:
Done in latest